### PR TITLE
Update CAPI to latest stable version

### PR DESCRIPTION
**Pull request by furkatgofurov7**
This uplifts `capi` and `controller-runtime` versions to latest as of now: **v0.4.0** and **v0.9.1** in the go modules to sync with CAPM3 master branch.  

**Note:** Those dependencies left behind without being updated in #37 since by the time of working on that PR there was no stable capi release but only beta version.

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
/retitle Update CAPI to latest stable version

File: go.mod [hunk 1]
```diff
@@ -3,22 +3,21 @@ module github.com/metal3-io/ip-address-manager
 go 1.16
 
 require (
-	cloud.google.com/go v0.75.0 // indirect
```
Comment (kashifest) [inline go.mod:6]: why is this getting removed?

Comment (furkatgofurov7) [reply go.mod:6]: Most probably because of the `go mod tidy`.

**Review by fmuyassarov — COMMENTED** [review]
/lgtm

**Review by kashifest — APPROVED** [review]

**Pull request merged**